declarations/linter.d.ts

```diff
@@ -0,0 +1,38 @@
+/** @typedef {import('eslint').ESLint} ESLint */
+/** @typedef {import('eslint').ESLint.Formatter} Formatter */
+/** @typedef {import('eslint').ESLint.LintResult} LintResult */
+/** @typedef {import('webpack').Compiler} Compiler */
+/** @typedef {import('./options').Options} Options */
+/** @typedef {import('./options').FormatterFunction} FormatterFunction */
+/**
+ * @param {Options} options
+ * @param {Compiler} compiler
+ * @returns {Promise<void>}
+ */
+export default function linter(
+  options: Options,
+  compiler: Compiler
+): Promise<void>;
+export type ESLint = import('eslint').ESLint;
+export type Formatter = import('eslint').ESLint.Formatter;
+export type LintResult = import('eslint').ESLint.LintResult;
+export type Compiler = import('webpack').Compiler;
```

Comment:
webpack types should be in `dependencies`, we will remove them when webpack@5 will have stable release (built-in types)

Reply:
Here we put it on devDependencies https://github.com/webpack-contrib/stylelint-webpack-plugin/pull/213#discussion_r440788832

Reply:
Now I'm in doubt, where should I stay lol

Reply:
Moved

Reply:
hm, ideally when you use webpack and typescript you should install `@webpack/types` too (otherwise why do you use typescript :smile: ), so I am wrong, we don't need `@types/webpack` in `dependencies`